src/EnergyPlus/SZVAVModel.cc

```diff
@@ -772,8 +773,11 @@ namespace SZVAVModel {
                         Par(9) = lowWaterMdot; // minCoilFluidFlow - low fan speed water flow rate > 0 if SAT limited
                         Par(12) = AirMassFlow; // sets air flow rate used when iterating on coil capacity
                         Par(13) = 0.0;         // other than 0 means to iterate on SA temperature
-
-                        TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilWaterFlowResidual, 0.0, 1.0, Par);
+                        if (SZVAVModel.HCoilType_Num == HCoil_Water || !HeatingLoad) {
+                            TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilWaterFlowResidual, 0.0, 1.0, Par);
+                        } else {
+                            TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilLoadResidual, 0.0, 1.0, Par);
+                        }
                         outletTemp = DataLoopNode::Node(OutletNode).Temp;
```

Comment:
Added SolveRoot() function calling for electric heating coil part-load ratio calculation using a different residual function.